drivers/hmis/app/graphql/mutations/update_form_definition.rb

```diff
@@ -35,5 +43,34 @@ def resolve(id:, input:)
         { errors: errors }
       end
     end
+
+    private
+
+    def recursively_transform(form_element)
+      # First drop unneeded keys and transform them all to snake case
+      converted = transform_hash(form_element)
+
+      # Then map through all the elements in the result object
+      converted.keys.each do |key|
+        if converted[key].is_a?(Array)
+          # If it's an array, recursively transform each element in the array
+          converted[key] = converted[key].map do |element|
+            recursively_transform(element)
+          end
```

Comment:
I don't think this will handle [an array of strings](https://github.com/greenriver/hmis-warehouse/blob/0f38f55bdaea72c641137925bfbfd0984bffe258/drivers/hmis/app/graphql/types/forms/enable_when.rb#L21) well. Maybe if  `recursively_transform` receives an `element` that is not a hash, it just returns the element directly?

Reply:
Oh true, thanks for pointing out the issue with arrays of strings. I updated it and added a regression test too. Hopefully this version is more robust!